bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,326 @@
+Galleon provisioning is operated inside the Maven plugin. There are 2 configuration ways:
+
+* Galleon configuration as Maven plugin configuration items:
+** A Galleon feature-pack location that identifies the server and version. This is required if no provisioning.xml file is provided nor feature-packs list.
+** A list of feature-packs (if no feature-pack location has been set).
+** Optionally, a list of layers to include.
+** Optionally, a list of layers to exclude.
+* Galleon provisioning.xml file located in <app src>/galleon/provisioning.xml. NB: Maven plugin configuration items (if set) override provisioning.xml file. 
+Using provisioning.xml file, custom feature-packs can be combined during the provisioning phase to provision modules, features, content (e.g.: DB drivers).
+
+If no Galleon layers are specified, then a configuration identical to the default _standalone-microprofile.xml_ is provisioned.
+
+===== CLI script execution
+
+When building the bootable JAR, a path to a set of CLI scripts can be provided to update the server configuration packaged in the bootable JAR. The CLI scripts are concatenated and executed once the server is provisioned and deployment copied into it (if any).
+Having multiple CLI scripts provides enough flexibility to reuse scripts in different contexts.
+The plugin can be configured with a CLI properties file to resolve properties present in CLI scripts. 
+This is similar to the _--properties=<properties file>_ option that you can provide to JBoss CLI command line. 
+This http://www.mastertheboss.com/jboss-server/jboss-script/using-properties-in-cli-scripts[article] covers the feature.
+
+The https://github.com/wildfly-extras/wildfly-jar-maven-plugin/releases/download/2.0.0.Alpha4/index.html[community documentation] contains details on how to setup CLI scripts and CLI properties file.
+
+===== User application
+
+The WAR or JAR main artifact the Maven project is building is deployed and zipped along with the server. 
+The plugin option root-url-path=true|false (true by default) allows to rename the war file to ROOT.war 
+(doesn’t apply to other packaging types). 
+
+NB: Application deployment doesn't rely on the deployment scanner. The scanner is not required, it is even suggested to exclude it, useless in a bootable JAR context.
+For an example of how to exclude the _deployment-scanner_ Galleon layer, you can check https://github.com/wildfly-extras/wildfly-jar-maven-plugin/blob/master/examples/jaxrs/pom.xml#L35[this example]
+
+In case the bootable is an “hollow JAR”, no deployment is present in the JAR.
+
+===== Hollow JAR
+
+No WAR/JAR/EAR file is copied into the server deployments. This is controlled by a plugin option.
+The hollow JAR doesn’t require the deployment-scanner to be provisioned in order for the deployment to be taken into account at runtime time. At runtime, deployment is copied to content dir, xml configuration is updated with the deployment.
+
```

Comment:
What about not passing `--deployment=myapp-wildfly.jar`. Would a user be able to deploy anyway later - e.g. via the CLI? I assume yes by reading the previous content but maybe in such case it would be better to state explicitly here that what described above is happening _if_ the option is passed in. Is this correct or am I missing something?

Reply:
Added a note for CLI deployment.